src/python.cc

```diff
@@ -466,88 +416,65 @@ ModelInstanceState::GetInputTensor(
 
   // Load raw data into input_tensor raw data.
   std::string* data_buffer = input_tensor->mutable_raw_data();
-  const void* input_buffer = nullptr;
-  uint64_t buffer_byte_size = 0;
-  TRITONSERVER_MemoryType input_memory_type = TRITONSERVER_MEMORY_CPU;
+  BackendMemory* input_memory;
+
   int64_t input_memory_type_id = 0;
-  for (size_t j = 0; j < input_buffer_count; ++j) {
-    RETURN_IF_ERROR(TRITONBACKEND_InputBuffer(
-        in, j, &input_buffer, &buffer_byte_size, &input_memory_type,
-        &input_memory_type_id));
-
-    if (input_memory_type == TRITONSERVER_MEMORY_GPU) {
-      RETURN_IF_ERROR(TRITONSERVER_ErrorNew(
-          TRITONSERVER_ERROR_UNSUPPORTED,
-          "failed to get input buffer in CPU memory"));
-    }
-    data_buffer->append((const char*)input_buffer, buffer_byte_size);
-  }
+  RETURN_IF_ERROR(BackendMemory::Create(
+      model_state_->TritonMemoryManager(),
+      {BackendMemory::AllocationType::CPU_PINNED_POOL,
```

Comment:
Pinned memory is "expensive" and so should only use it if going to copy to/from GPU.